src/poppler/document.py

```diff
@@ -58,24 +59,26 @@ def __init__(self, poppler_document, data=None):
             raise ValueError("Corrupted or invalid PDF document")
 
         self._document = poppler_document
-        self._data = data
+        self._data: bytes = data
```

Comment:
I would not annotate this internal variable. It is not part of the API, and its sole reason to be there is to prevent data being garbage collected when you initialize the document from bytes.

Reply:
done + adding a comment about the purpose